staging/src/k8s.io/client-go/util/workqueue/delaying_queue_test.go

```diff
@@ -214,6 +215,18 @@ func TestCopyShifting(t *testing.T) {
 	}
 }
 
+func BenchmarkDelayingQueue_AddAfter(b *testing.B) {
+	r := rand.New(rand.NewSource(time.Now().Unix()))
+
+	fakeClock := clock.NewFakeClock(time.Now())
+	q := newDelayingQueue(fakeClock, "")
+
+	for n := 0; n < b.N; n++ {
+		data := fmt.Sprintf("%d", n)
+		q.AddAfter(data, time.Duration(r.Int63n(int64(10*time.Minute))))
```

Comment:
I'd like to see removal exercised as well... the new impl must sweep every pending item on every Pop() to update indices...